sunpy/net/dataretriever/sources/aia_synoptic.py

```diff
@@ -0,0 +1,195 @@
+    known_wavelengths = [94, 131, 171, 193, 211, 304, 335, 1600, 1700, 4500]
+
+    required = {a.Time, a.Instrument, a.Level}
+    supported = required + {a.Sample, a.Wavelength, a.ExtentType}
+
+    @property
+    def info_url(self):
+        return self.baseurl
+
+    @classmethod
+    def register_values(cls):
+        adict = {
+            a.Instrument: [("AIA", "Data from the Atmospheric Imaging Assembly instrument.")],
+            a.Physobs: [
+                ("intensity", "Brightness or intensity of the solar atmosphere at different wavelengths.")
+            ],
+            a.Source: [("SDO", "The Solar Dynamics Observatory.")],
+            a.Wavelength: [(f"{wv:04d}", f"{wv} Å") for wv in cls.known_wavelengths],
+            a.Provider: [("JSOC", "Joint Science Operations Center at Stanford.")],
+            a.Level: [("synoptic", "Level 1.5 data processed for quicker analysis.")],
+            a.ExtentType: [("synoptic", "Level 1.5 data processed for quicker analysis.")],
+        }
+        return adict
+
+    def _can_handle_query(self, *query):
+        """
+        Determines whether this client can handle the given query.
+        """
+        required_attrs = {a.Instrument, a.Level}
+        supported_attrs = {a.Time, a.Instrument, a.Sample, a.Level, a.Wavelength, a.ExtentType}
+
+        # Check if all required attributes are present in the query
+        query_attrs = set(type(attr) for attr in query)
+        if not required_attrs.issubset(query_attrs):
+            return False
+
+        # Ensure there are no unsupported attributes in the query
+        if not query_attrs.issubset(supported_attrs):
+            return False
+
+        return True
+
+    def search(self, *args, **kwargs):
+        """
+        Perform a search for AIA synoptic data.
+        """
+        matchdict = self._get_match_dict(*args, **kwargs)
+        time_range = matchdict["time"]
+        start_time = time_range.start.datetime
+        end_time = time_range.end.datetime
+        sample = matchdict.get("sample", None)  # Get the sampling interval if provided
+        sample_minutes = (
+            sample.to(u.min).value if sample is not None else 2
+        )  # Default to 2-minute intervals if not provided
+
+        # List of wavelengths to search
+        wavelengths = [matchdict["wavelength"]] if "wavelength" in matchdict else self.known_wavelengths
+
+        # Calculate the number of steps based on the sample interval
+        total_minutes = (end_time - start_time).total_seconds() / 60
+        num_steps = int(np.ceil(total_minutes / sample_minutes))
+
+        # Generate the array of times
+        time_steps = [
+            start_time + timedelta(minutes=i * sample_minutes)
+            for i in range(num_steps + 1)
+            if start_time + timedelta(minutes=i * sample_minutes) <= end_time
+        ]
+
+        # Generate all possible file URLs based on the time steps and wavelengths
```

Comment:
Should this be a scarper client instead of just constructing the URLs?

Reply:
Well, I started off that way and for queries with large ranges it was literally checking millions of links for compliance with the query. This way was 100x faster or something.

Reply:
Good point, I didn't consider the speed angle.